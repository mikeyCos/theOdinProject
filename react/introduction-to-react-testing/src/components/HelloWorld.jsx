export default function HelloWorld() {
  return (
    <section id="hello-world">
      <h2>Hello World</h2>
      <h3>testing</h3>
    </section>
  );
}
