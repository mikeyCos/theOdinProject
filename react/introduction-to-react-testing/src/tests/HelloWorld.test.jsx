import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import HelloWorld from "../components/HelloWorld";

describe("HelloWorld component", () => {
  it("Renders correct HelloWorld", () => {
    const { container } = render(<HelloWorld />);
    expect(container).toMatchSnapshot();
  });

  it("Renders correct headings", () => {
    const { getAllByRole } = render(<HelloWorld />);
    const headings = getAllByRole("heading").map(
      (heading) => heading.textContent
    );

    // Headings should have textContent of "Hello World" and "testing"
    expect(
      headings.reduce((accumulator, currentHeading) => {
        return accumulator && /(hello world)|(testing)/i.test(currentHeading);
      }, true)
    ).toBeTruthy();
  });
});
