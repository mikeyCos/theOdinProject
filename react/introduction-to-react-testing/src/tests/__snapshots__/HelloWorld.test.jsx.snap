// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HelloWorld component > Renders correct HelloWorld 1`] = `
<div>
  <section
    id="hello-world"
  >
    <h2>
      Hello World
    </h2>
    <h3>
      testing
    </h3>
  </section>
</div>
`;
